lib/manager/nuget/extract.ts

```diff
@@ -1,10 +1,50 @@
+import * as findUp from 'find-up';
+import * as path from 'path';
+import { XmlDocument } from 'xmldoc';
+import { readFileSync } from 'fs';
 import { logger } from '../../logger';
 import { get } from '../../versioning';
 import { PackageDependency, ExtractConfig, PackageFile } from '../common';
 import * as semverVersioning from '../../versioning/semver';
 import * as datasourceNuget from '../../datasource/nuget';
 import { SkipReason } from '../../types';
 
+function determineRegistryUrls(
+  packageFile: string,
+  localDir: string
+): string[] {
+  const registryUrls = [datasourceNuget.getDefaultFeed()];
+  const nuGetConfigPath = findUp.sync('NuGet.config', {
+    cwd: path.dirname(path.join(localDir, packageFile)),
+    type: 'file',
+  });
+  if (nuGetConfigPath && nuGetConfigPath.startsWith(localDir)) {
+    logger.info(`found NuGet.config at '${nuGetConfigPath}'`);
+    const nuGetConfig = new XmlDocument(
+      readFileSync(nuGetConfigPath).toString()
```

Comment:
This should also be an await/promise

Reply:
Done with 5dd788c881168b15294db1569eea037b4753ea30